### PR TITLE
Be refactor new fix test bug

### PR DESCRIPTION
**Pull request by kangpinghuang**
1. fix segment reader DCHECK bug
2. fix create rowset reader nullptr bug

### CODE REVIEW

File: be/src/agent/task_worker_pool.cpp [hunk 1]
```diff
@@ -626,6 +626,10 @@ void TaskWorkerPool::_alter_tablet(
             finish_tablet_infos.push_back(tablet_info);
         }
     }
+    
+    for (auto& tablet_info : finish_tablet_infos) {
```
Comment (chaoyli) [inline be/src/agent/task_worker_pool.cpp]: Debug info is not necessary.

File: be/src/agent/task_worker_pool.cpp [hunk 2]
```diff
@@ -626,6 +626,10 @@ void TaskWorkerPool::_alter_tablet(
...
+        LOG(INFO) << "finish tablet infos:" <<  apache::thrift::ThriftDebugString(tablet_info);
+    }
```
Comment (yiguolei) [inline be/src/agent/task_worker_pool.cpp]: It should be debug or vlog or there will be too much logs.

File: be/src/olap/reader.cpp [hunk 3]
```diff
@@ -467,7 +467,7 @@ OLAPStatus Reader::_capture_rs_readers(const ReaderParams& read_params) {
         rs_readers = &read_params.rs_readers;
     } else {
         _tablet->obtain_header_rdlock();
-        _tablet->capture_rs_readers(_version, &_own_rs_readers);
+        RETURN_NOT_OK(_tablet->capture_rs_readers(_version, &_own_rs_readers));
```
Comment (chaoyli) [inline be/src/olap/reader.cpp]: lock leak

File: be/src/olap/rowset/alpha_rowset_reader.cpp [hunk 4]
```diff
@@ -234,6 +234,7 @@ OLAPStatus AlphaRowsetReader::_init_column_datas(RowsetReaderContext* read_conte
         _key_range_size = read_context->lower_bound_keys->size();
     }
 
+    LOG(INFO) << "segment groups size:" << _segment_groups.size();
```
Comment (chaoyli) [inline be/src/olap/rowset/alpha_rowset_reader.cpp]: Debug info is not necessary.

File: be/src/olap/rowset/alpha_rowset_reader.cpp [hunk 5]
```diff
@@ -311,8 +317,9 @@ OLAPStatus AlphaRowsetReader::_init_column_datas(RowsetReaderContext* read_conte
             }
         } else {
             status = new_column_data->get_first_row_block(&row_block);
-            if (status != OLAP_SUCCESS) {
-                LOG(WARNING) << "get first row block failed";
+            LOG(INFO) << "get_first_row_block status:" << status;
```
Comment (chaoyli) [inline be/src/olap/rowset/alpha_rowset_reader.cpp]: Debug info is not necessary.

File: be/src/olap/reader.cpp [hunk 6]
```diff
@@ -467,7 +467,7 @@ OLAPStatus Reader::_capture_rs_readers(const ReaderParams& read_params) {
...
         _tablet->release_header_lock();
```
Comment (yiguolei) [inline be/src/olap/reader.cpp]: if not return ok, it will return if own_res_readers's size < 1. Is that ok?

**Review by chaoyli — CHANGES_REQUESTED** [review]

File: be/src/olap/rowset/alpha_rowset_reader.cpp [hunk 7]
```diff
@@ -285,6 +286,11 @@ OLAPStatus AlphaRowsetReader::_init_column_datas(RowsetReaderContext* read_conte
         _row_cursors.push_back(row_cursor);
 
         RowBlock* row_block = nullptr;
+        if (segment_group->empty()) {
+            LOG(INFO) << "segment group is empty";
```
Comment (imay) [inline be/src/olap/rowset/alpha_rowset_reader.cpp]: useless log

File: be/src/agent/task_worker_pool.cpp (hunk 1 above)
Comment (kangpinghuang) [reply be/src/agent/task_worker_pool.cpp]: OK

File: be/src/agent/task_worker_pool.cpp (hunk 2 above)
Comment (kangpinghuang) [reply be/src/agent/task_worker_pool.cpp]: ok

File: be/src/olap/reader.cpp (hunk 3 above)
Comment (kangpinghuang) [reply be/src/olap/reader.cpp]: ok

File: be/src/olap/reader.cpp (hunk 6 above)
Comment (kangpinghuang) [reply be/src/olap/reader.cpp]: Here return not ok means that the reader init failed and the reading will return failed directly. I think it is a right process.

File: be/src/agent/task_worker_pool.cpp [hunk 8]
```diff
@@ -633,8 +631,8 @@ void TaskWorkerPool::_alter_tablet(
         error_msgs.push_back(process_name + " success");
         task_status.__set_status_code(TStatusCode::OK);
     } else if (status == DORIS_TASK_REQUEST_ERROR) {
-        OLAP_LOG_WARNING("alter table request task type invalid. "
-                         "signature: %ld", signature);
+        LOG(WARNING) << "alter table request task type invalid. "
+                         "signature:" << signature;
         error_msgs.push_back("alter table request new tablet id or schema count invalid.");
```
Comment (yiguolei) [inline be/src/agent/task_worker_pool.cpp]: missed << ?

**Review by yiguolei — APPROVED** [review]

File: be/src/olap/reader.cpp [hunk 9]
```diff
@@ -467,8 +467,11 @@ OLAPStatus Reader::_capture_rs_readers(const ReaderParams& read_params) {
         rs_readers = &read_params.rs_readers;
     } else {
         _tablet->obtain_header_rdlock();
-        _tablet->capture_rs_readers(_version, &_own_rs_readers);
+        OLAPStatus status = _tablet->capture_rs_readers(_version, &_own_rs_readers);
         _tablet->release_header_lock();
+        if (status != OLAP_SUCCESS) {
```
Comment (chaoyli) [inline be/src/olap/reader.cpp]: RETUNR_NOT_OK(status)

**Review by chaoyli — CHANGES_REQUESTED** [review]

**Comment by kangpinghuang** [comment]
resolve ISSUE#771

**Pull request merged**